agent-shell.el

```diff
@@ -520,6 +592,25 @@ Flow:
                                                "│ changes │\n"
                                                "╰─────────╯\n\n" diff-text)
                                      output)))
+                   ;; Log tool call to transcript when completed or failed
```

Comment:
Let's add a helper we can use here... Maybe `agent-shell--make-transcript-tool-call-entry`, so it looks something like:

(agent-shell--append-to-transcript
  :file-path ...
  :text (agent-shell--make-transcript-tool-call-entry ...)